### PR TITLE
Accelerate with copilot

### PR DESCRIPTION
**Pull request by andmariano**
This pull request introduces a major enhancement to the Mergington High School Activities app by adding participant management features, improving the UI for activity participation, and introducing comprehensive automated tests for the API. The most important changes include the ability to unregister participants from activities, a new participant list UI with removal controls, and a robust test suite to ensure API correctness.

**Backend Features:**

- Added endpoint to unregister a participant from an activity, with input validation and error handling (`src/app.py`).
- Prevented duplicate participant signups for the same activity (`src/app.py`).

**Frontend/UI Improvements:**

- Displayed a dynamic list of participants for each activity, including a delete button for each participant (`src/static/app.js`).
- Enabled participant removal via the UI, with confirmation dialogs and real-time updates (`src/static/app.js`).
- Refreshed the activities list after participant signup or removal to reflect changes immediately (`src/static/app.js`).
- Styled the participant list and delete button for better usability and appearance (`src/static/styles.css`).

**Testing and Quality Assurance:**

- Added a comprehensive test suite covering all major API endpoints, including signup, unregister, activity retrieval, edge cases, and capacity checks (`tests/test_api.py`, `tests/__init__.py`). [[1]](diffhunk://#diff-3451cc474d5698222168f66a293339a6bd296ee2c56f56409866bbae99917a79R1-R254) [[2]](diffhunk://#diff-001e61d97d9dee27d0c5aa1f23ba6c1972cd8a2e8320bac839656b0ab935b84dR1)

### CODE REVIEW

File: src/static/app.js [hunk 1]
```diff
@@ -35,12 +71,62 @@ document.addEventListener("DOMContentLoaded", () => {
         option.textContent = name;
         activitySelect.appendChild(option);
```
[the remark this answers is not in the document]
Comment (andmariano) [reply src/static/app.js:72]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/andmariano/flai-workshop-github-copilot-100/pull/2#discussion_r2823311416)

File: src/static/app.js [hunk 2]
```diff
@@ -19,12 +19,48 @@ document.addEventListener("DOMContentLoaded", () => {
...
+        let participantsList = '';
+        if (details.participants.length > 0) {
+          const participantItems = details.participants
+            .map(email => `
+              <li>
+                <span class="participant-email">${email}</span>
+                <button class="delete-btn" data-activity="${name}" data-email="${email}" title="Remove participant">
+                  <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
+                    <polyline points="3 6 5 6 21 6"></polyline>
+                    <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path>
```
[the remark this answers is not in the document]
Comment (andmariano) [reply src/static/app.js:33]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/andmariano/flai-workshop-github-copilot-100/pull/2#discussion_r2823311481)

File: src/static/app.js [hunk 3]
```diff
@@ -19,12 +19,48 @@ document.addEventListener("DOMContentLoaded", () => {
...
+        
+        // Create participants list
+        let participantsList = '';
+        if (details.participants.length > 0) {
+          const participantItems = details.participants
+            .map(email => `
+              <li>
+                <span class="participant-email">${email}</span>
+                <button class="delete-btn" data-activity="${name}" data-email="${email}" title="Remove participant">
+                  <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
```
[the remark this answers is not in the document]
Comment (andmariano) [reply src/static/app.js:31]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/andmariano/flai-workshop-github-copilot-100/pull/2#discussion_r2823311523)
